snowflake_view_grant/main.tf.json

```diff
@@ -0,0 +1,70 @@
+{
+  "//": "NOTE: Generated by scripts/snowflake_generate_grant_all.\n\tChanges made directly to this file will be overwritten.\n\tMake improvements there so everyone can benefit.\n  The reason this module exists is that the provider only supports one grant resource per (database_name, schema_name, table_name, on_future, with_grant_option) tuple.\n\tFor example, if you used this module to grant an ALL privilege to a role you couldn't grant a subset of the ALL privs to another role.",
+  "variable": {
+    "database_name": {
+      "type": "string",
+      "description": "The name of the database containing the current or future views on which to grant privileges.",
+      "default": null
+    },
+    "on_future": {
+      "type": "bool",
+      "description": "When this is set to true and a schema_name is provided, apply this grant on all future views in the given schema. When this is true and no schema_name is provided apply this grant on all future views in the given database. The view_name and shares fields must be unset in order to use on_future.",
+      "default": null
+    },
+    "per_privilege_grants": {
+      "type": "map(object({shares : list(string),roles : list(string)}))",
+      "description": "A map of privileges to authorized roles and shares. Privileges must be UPPER case.\n  This allows you to authorize extra roles/shares for specific privileges.",
+      "default": {}
+    },
+    "roles": {
+      "type": "set(string)",
+      "description": "Grants privilege to these roles.",
+      "default": null
+    },
+    "schema_name": {
+      "type": "string",
+      "description": "The name of the schema containing the current or future views on which to grant privileges.",
+      "default": null
+    },
+    "shares": {
+      "type": "set(string)",
+      "description": "Grants privilege to these shares (only valid if on_future is unset).",
+      "default": null
+    },
+    "view_name": {
+      "type": "string",
+      "description": "The name of the view on which to grant privileges immediately (only valid if on_future is unset).",
+      "default": null
+    },
+    "with_grant_option": {
+      "type": "bool",
+      "description": "When this is set to true, allows the recipient role to grant the privileges to other roles.",
+      "default": null
+    }
+  },
+  "locals": {
+    "privileges": [
+      "SELECT"
+    ]
+  },
+  "resource": {
+    "snowflake-view-grant-all": {
+      "all": {
```

Comment:
⚠️ **[tflint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Missing version constraint for provider "snowflake-view-grant-all" in "required_providers"